src/Models/AzureComponentConfig.ts

```diff
@@ -107,11 +105,10 @@ export class AzureConfigFileHandler {
     }
   }
 
-  getComponentIndexById(id: string) {
+  async getComponentIndexById(id: string) {
     try {
-      const azureConfigs =
-          JSON.parse(fs.readFileSync(this.configFilePath, 'utf8')) as
-          AzureConfigs;
+      const azureConfigContent = await FileUtility.readFile(ScaffoldType.local, this.configFilePath, 'utf8');

```

Comment:
why using scaffoldtype.local here and workspace in async getSortedComponents

Reply:
Same as above.